src/services/yolo_detection_network.py

```diff
@@ -56,17 +94,36 @@ async def capture_all_from_camera(
         extra: Optional[Mapping[str, Any]] = None,
         timeout: Optional[float] = None,
     ) -> CaptureAllResult:
-        raise NotImplementedError
-
-    async def get_object_point_clouds(
-        self,
-        camera_name: str,
-        *,
-        extra: Optional[Dict[str, Any]] = None,
-        timeout: Optional[float] = None,
-        **kwargs,
-    ) -> List[PointCloudObject]:
-        raise NotImplementedError
+        if camera_name == "" or camera_name is None:
+            self.logger.warn(
+                f"camera_name arg was unspecified. Defaulting to '{self.cam_name}'"
+            )
+        elif camera_name != self.cam_name:
+            raise ViamError(
+                f'Requested camera name "{camera_name}" does not match configured OAK camera "{self.cam_name}".'
+            )
+
+        if return_classifications:
+            raise ViamError(
+                "Classifications are not supported in the YOLO detection network service."
+            )
+        if return_object_point_clouds:
+            raise ViamError(
+                "Object point clouds are not supported in the YOLO detection network service."
+            )
+        if return_image == False and return_detections == False:
+            raise ViamError("Please request either return_image or return_detections.")
+
+        mapping = await self.cam.do_command(
+            {
+                "cmd": YDN_CAPTURE_ALL,
+                "return_detections": return_detections,
+                "return_image": return_image,
+                "sender_id": self.ydn_service_id.hex,
+                "sender_name": self.ydn_service_name,
+            }
+        )
+        return convert_capture_all_dict_into_capture_all(mapping)
 
     async def get_detections(
```

Comment:
I implemented this because it's easier to test get_detections (thanks the RC card) than it is to test get_detections_from_cam